Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM golang:1.14.2-alpine AS builder
+
+RUN apk add --update --no-cache ca-certificates bash make gcc musl-dev git openssh wget curl bzr
+
+WORKDIR /go/src/mattermod
+
+COPY go.mod .
```

Comment:
Why don't do `COPY . .` and then `RUN go mod download`?

Reply:
because frequency of changes in `go.mod` and `go.sum` are way lower than the rest of the codebase, so with this approach these three layers (`COPY go.mod`, `COPY go.sum` and `RUN go mod download`) do get cached properly and **only** will re-run again if there's actual changes in `go.mod` or `go.sum` or both.

Reply:
Makes total sense